exercises/nucleotide-count/.meta/gen.go

```diff
@@ -0,0 +1,84 @@
+package main
+
+import (
+	"log"
+	"sort"
+	"strconv"
+	"strings"
+	"text/template"
+
+	"../../../gen"
+)
+
+func main() {
+	t, err := template.New("").Parse(tmpl)
+	if err != nil {
+		log.Fatal(err)
+	}
+	var j js
+	if err := gen.Gen("nucleotide-count", &j, t); err != nil {
+		log.Fatal(err)
+	}
+}
+
+// The JSON structure we expect to be able to unmarshal into
+type js struct {
+	exercise string
+	version  string
+	Cases    []struct {
+		Description string
+		Cases       []OneCase
+	}
+}
+
+// OneCase represents each test case
+type OneCase struct {
+	Description string
+	Property    string
+	Strand      string
+	Expected    map[string]interface{}
+}
+
+// ErrorExpected returns true if an error should be raised
+func (c OneCase) ErrorExpected() bool {
+	_, exists := c.Expected["error"]
+	return exists
+}
+
+// SortedMapString collects key:values for a map in sorted order
+func (c OneCase) SortedMapString() string {
+	strs := make([]string, 0, len(c.Expected))
+	for k, v := range c.Expected {
+		switch t := v.(type) {
+		case float64:
+			strs = append(strs, `'`+k+`': `+strconv.FormatFloat(t, 'f', -1, 64))
+		default:
+		}
+
```

Comment:
rm

Reply:
Yeah,
> default:

That line is unnecessary. Can remove it.

Reply:
ok .. I made it a little more useful :) 

in the meantime the problem specification json structure has changed, so I updated also the generator